src/leap/mail/adaptors/soledad.py

```diff
@@ -527,26 +530,41 @@ def create(self, store, notify_just_mdoc=False):
         leap_assert(self.fdoc.doc_id is None,
                     "Cannot create: fdoc has a doc_id")
 
+        def unblock_pending_insert(result):
+            msgid = self.hdoc.headers.get('Message-Id', None)
+            try:
+                d = pending_inserts_dict[msgid]
+                d.callback(msgid)
+            except KeyError:
+                pass
+            return result
+
         # TODO check that the doc_ids in the mdoc are coherent
-        d = []
+        self.d = []
```

Comment:
Why do `d` need to become a member of the object? I don't see it used anywhere else.

I don't like having an object member with so sort name, it's even confusing as I will expect a deferred and in reality is a list of deferreds. Can we give it a name that is more meaningful.